models/nlp/classification/huggingface/inference.py

```diff
@@ -0,0 +1,56 @@
+import zipfile
+
+import torch
+import valohai
+from transformers import AutoModelForSequenceClassification, AutoTokenizer
+
+from utils.serializers import get_serializer
+
+
+def load_model(model_path):
+    tokenizer = AutoTokenizer.from_pretrained(model_path)
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    model = AutoModelForSequenceClassification.from_pretrained(model_path)
+    model.to(device)
+    return tokenizer, model, device
+
+
+def predict(text, tokenizer, model, device):
+    inputs = tokenizer(text, return_tensors="pt", padding=True, truncation=True)
+    inputs = {key: tensor.to(device) for key, tensor in inputs.items()}
+    outputs = model(**inputs)
+    logits = outputs.logits
+    predictions = torch.argmax(logits, dim=-1)
+    return predictions.item()
+
+
+def main():
+    model_path = valohai.inputs("model").path(process_archives=False)
+    if model_path.endswith(".zip"):
+        with zipfile.ZipFile(model_path, "r") as zip_ref:
+            zip_ref.extractall("/tmp/model")  # noqa: S108
+    else:
+        print(f"Model {model_path} is not a zip file. Aborting.")
+        return
+
+    tokenizer, model, device = load_model("/tmp/model")  # noqa: S108
+    data_path = valohai.inputs("data").path("*.txt")
+    rows = open(data_path).readlines()
+
+    results = []
+    for index, row in enumerate(rows):
```

Comment:
Use `with` to open the file, and don't `readlines()` (because it will read the file into memory); instead just iterate over the lines. (Of course this means you won't know the total count of rows – if that's critical to know, you need to use `.readlines()`.)



```suggestion
with open(data_path) as f:
    results = []
    for index, row in enumerate(f):
```